File: cakk-client/src/main/java/com/cakk/client/web/AppleAuthClient.java
package com.cakk.client.web;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.springframework.web.client.RestClient;

import com.cakk.client.vo.OidcPublicKeyList;

@Component
public class AppleAuthClient {

	private final RestClient restClient;

	private final String publicKeyUrl;

	public AppleAuthClient(
		RestClient restClient,
		@Value("${oauth.apple.public-key-url}") String publicKeyUrl
	) {
		this.restClient = restClient;
		this.publicKeyUrl = publicKeyUrl;
	}

	public OidcPublicKeyList getPublicKeys() {
		return restClient.get()
			.uri(publicKeyUrl)
			.retrieve()
			.body(OidcPublicKeyList.class);
	}
}

File: cakk-external/src/main/kotlin/com/cakk/external/client/AppleAuthClient.kt
package com.cakk.external.client

import com.cakk.common.enums.ReturnCode
import com.cakk.common.exception.CakkException
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import org.springframework.web.client.RestClient

import com.cakk.external.vo.key.OidcPublicKeyList

@Component
class AppleAuthClient(
	private val restClient: RestClient,
	@Value("\${oauth.apple.public-key-url}")
	private val publicKeyUrl: String
) {

    fun getPublicKeys(): OidcPublicKeyList {
		return restClient.get()
			.uri(publicKeyUrl)
			.retrieve()
			.body(OidcPublicKeyList::class.java)
			?: throw CakkException(ReturnCode.EXTERNAL_SERVER_ERROR)
	}
}
